src/dftbp/dftbplus/parser.F90

```diff
@@ -3925,7 +3922,13 @@ subroutine readOptions(node, ctrl)
     if (ctrl%tReadChrg) then
       call getChildValue(node, "ReadChargesAsText", ctrl%tReadChrgAscii, .false.)
     end if
-    call getChildValue(node, "WriteChargesAsText", ctrl%tWriteChrgAscii, .false.)
+
+    call getChildValue(node, "WriteCharges", ctrl%tWriteCharges, .true.)
+    call getChildValue(node, "WriteChargesAsText", ctrl%tWriteChrgAscii, .false., child=child)
+    if (ctrl%tWriteChrgAscii .and. (.not. ctrl%tWriteCharges)) then
+      call detailedError(child, "Charges requested in Ascii format, but writing out charges is&
+          & disabled (contradictory input).")
+    end if
```

Comment:
```suggestion
    call getChildValue(node, "WriteCharges", ctrl%tWriteCharges, .true.)
    if (ctrl%tWriteCharges) then
      call getChildValue(node, "WriteChargesAsText", ctrl%tWriteChrgAscii, .false.)
    end if
```

Reply:
@bhourahine Like this?

Reply:
@vanderhe sure, you could include the ascii if test that follows it as well, but that's harmless as is.